subsys/bluetooth/host/crypto.c

```diff
@@ -111,6 +136,88 @@ int bt_rand(void *buf, size_t len)
 }
 #endif /* CONFIG_BT_HOST_CRYPTO_PRNG */
 
+#if defined(CONFIG_BT_USE_PSA_API)
+int bt_encrypt_le(const uint8_t key[16], const uint8_t plaintext[16],
+		  uint8_t enc_data[16])
+{
+	psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+	psa_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+	psa_status_t status, destroy_status;
+	size_t out_len;
+	uint8_t tmp[16];
+
+	CHECKIF(key == NULL || plaintext == NULL || enc_data == NULL) {
+		return -EINVAL;
+	}
+
+	LOG_DBG("key %s", bt_hex(key, 16));
+	LOG_DBG("plaintext %s", bt_hex(plaintext, 16));
+
+	sys_memcpy_swap(tmp, key, 16);
+
+	psa_set_key_type(&attr, PSA_KEY_TYPE_AES);
+	psa_set_key_bits(&attr, 128);
+	psa_set_key_usage_flags(&attr, PSA_KEY_USAGE_ENCRYPT);
+	psa_set_key_algorithm(&attr, PSA_ALG_SHA_256);
+	if (psa_import_key(&attr, tmp, 16, &key_id) != PSA_SUCCESS) {
+		return -EINVAL;
+	}
+
+	sys_memcpy_swap(tmp, plaintext, 16);
+
+	status = psa_cipher_encrypt(key_id, PSA_ALG_SHA_256, tmp, 16,
```

Comment:
Seems like you still didn't fix this, i.e. it's referencing SHA_256 when it should be using AES_128.

Reply:
Ops, sorry, you are absolutely right. This clearly means that the test that I added using PSA APIs is not checking this path. I will investigate and add another one.